> Build error on EWS for watch and playstation
> 
> ```
> /Volumes/Data/worker/watchOS-11-Build-EWS/build/Source/JavaScriptCore/runtime/Butterfly.h:244:31: error: inline function 'JSC::Butterfly::clearOptimalVectorLengthGap' is not defined [-Werror,-Wundefined-inline]
>   244 |     ALWAYS_INLINE static void clearOptimalVectorLengthGap(IndexingType, Butterfly*, unsigned optimalVectorLength, unsigned vectorLength);
>       |                               ^
> In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/WebKitBuild/Release-watchos/DerivedSources/JavaScriptCore/unified-sources/UnifiedSource1.cpp:1:
> In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/WebKitBuild/Release-watchos/DerivedSources/JavaScriptCore/BytecodeDumperGenerated.cpp:31:
> In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/WebKitBuild/Release-watchos/DerivedSources/JavaScriptCore/BytecodeStructs.h:31:
> In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/Source/JavaScriptCore/bytecode/ArrayAllocationProfile.h:29:
> /Volumes/Data/worker/watchOS-11-Build-EWS/build/Source/JavaScriptCore/runtime/JSArray.h:254:20: note: used here
>   254 |         Butterfly::clearOptimalVectorLengthGap(indexingType, butterfly, vectorLength, 0);
>       |                    ^
> 1 error generated.
> ```

Fixed.

--- a/Source/JavaScriptCore/generator/DSL.rb
+++ b/Source/JavaScriptCore/generator/DSL.rb
@@ -263,6 +263,7 @@ EOF
 #include "config.h"
 #include "BytecodeDumper.h"
 
+#include "ButterflyInlines.h"
 #include "BytecodeStructs.h"
 
 namespace JSC {

--- a/Source/JavaScriptCore/runtime/JSArray.h
+++ b/Source/JavaScriptCore/runtime/JSArray.h
@@ -251,10 +251,7 @@ inline JSArray* JSArray::tryCreate(VM& vm, Structure* structure, unsigned initia
         butterfly = Butterfly::fromBase(temp, 0, outOfLineStorage);
         butterfly->setVectorLength(vectorLength);
         butterfly->setPublicLength(initialLength);
-        if (hasDouble(indexingType))
-            clearArray(butterfly->contiguousDouble().data(), vectorLength);
-        else
-            clearArray(butterfly->contiguous().data(), vectorLength);
+        Butterfly::clearOptimalVectorLengthGap(indexingType, butterfly, vectorLength, 0);
     } else {
         ASSERT(
             indexingType == ArrayWithSlowPutArrayStorage

